### PR TITLE
[Phi] Move softmax with cross entropy kernel into phi

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

Function optimization
### PR changes

OPs
### Describe

[Phi] Move softmax with cross entropy kernel into phi 

1. softmax_with_cross_entropy op现在对应的Python API是cross_entropy，因此kernel文件命名为cross_entropy_kernel.*
2. softmax_with_cross_entropy op名不符实，现在它的行为是一个可以选择是否先进行softmax计算，再进行cross_entropy计算，cross_entropy计算是一定有的，而softmax不一定计算，因此将kernel更名为cross_entropy_with_softmax，实际上这也只是一种特殊的cross_entropy运算而已
3. 前向CPU kernel有存在显而易见的冗余代码，整合了重复代码，进行了简化封装
4. 内部调用函数保持原命名，未进行更改，有必要后续再进行，仅kernel名进行了调整

### CODE REVIEW

**Review by zyfncg — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
